vhdbuilder/packer/build-performance/build-performance-funcs.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+script_start_stopwatch=$(date +%s)
+section_start_stopwatch=$(date +%s)
+SCRIPT_NAME=$(basename $0 .sh)
+SCRIPT_NAME="${SCRIPT_NAME//-/_}"
+declare -A benchmarks=()
+declare -a benchmarks_order=()
+VHD_BUILD_PERF_DATA=${BUILD_PERF_DATA_FILE}
+
+check_array_size() {
```

Comment:
we already define these here: https://github.com/Azure/AgentBaker/blob/master/parts/linux/cloud-init/artifacts/cse_helpers.sh#L500-L548

could we upload this file to the packer VM and source them within install-deps as well? that way we won't have to worry about updating code in multiple places

Reply:
Do you mean remove these functions from `cse_helpers.sh` and source them to the install scripts from this file only? I like that idea and was initially going in that direction. 

The only reason I ended up not wanting to remove them from `cse_helpers.sh` was because I thought they could be useful for node bootstrapping benchmarking. But looking around yesterday, it seems we already have some infrastructure for logging during the bootstrapping process. 

Unfortunately, sourcing `cse_helpers.sh` to these files (convert, test, scan, etc) fails the step because of the variable assignments, so its a bit of a dilemma. How would you like me to proceed? @cameronmeissner

Reply:
should we decouple the variable assignment and the helper functions into two files ? this way one can source the helper functions only ?

Reply:
I agree though this is some pretty bad code duplication :(

Reply:
Is it possible to just source the functions into cse_helpers from this new file...? That way they are available to scripts that run after the packer vm is deprovisioned, and still available in cse_helpers for node bootstrap benchmarking later on.

@djsly @cameronmeissner